external/fv3fit/fv3fit/reservoir/readout.py

```diff
@@ -20,36 +27,116 @@ def square_even_terms(v: np.ndarray, axis=1) -> np.ndarray:
 class ReservoirComputingReadout:
     """Readout layer of the reservoir computing model
 
-    linear_regressor: a sklearn Ridge regreesor
-    square_half_hidden_state: if True, square even terms in the reservoir state
-        before it is used as input to the regressor's .fit and .predict methods
-        This option was found to be important for skillful predictions in
-        Wikner+2020 (https://doi.org/10.1063/5.0005541)
+    linear_regressor: a sklearn Ridge regressor
+    square_half_hidden_state: if True, square even terms in the reservoir
+        state before it is used as input to the regressor's .fit and
+        .predict methods. This option was found to be important for skillful
+        predictions in Wikner+2020 (https://doi.org/10.1063/5.0005541)
     """
 
     def __init__(
-        self, linear_regressor: Ridge, square_half_hidden_state: bool = False,
+        self,
+        hyperparameters: ReadoutHyperparameters,
+        coefficients: Optional[np.ndarray] = None,
+        intercepts: Optional[np.ndarray] = None,
     ):
-        self.linear_regressor = linear_regressor
-        self.square_half_hidden_state = square_half_hidden_state
+        self.hyperparameters = hyperparameters
+        self.coefficients = coefficients
+        self.intercepts = intercepts
+        self.square_half_hidden_state = hyperparameters.square_half_hidden_state
 
     def fit(self, res_states: np.ndarray, output_states: np.ndarray) -> None:
-        if self.square_half_hidden_state:
+        if self.coefficients or self.intercepts:
+            raise ValueError(
+                "Readout has already been fit and has coefficients and intercept "
+                "values. Fit method can only be called if readout is yet fit."
+            )
+        linear_regressor = Ridge(**self.hyperparameters.linear_regressor_kwargs)
+        if self.square_half_hidden_state is True:
             res_states = square_even_terms(res_states, axis=1)
-        self.linear_regressor.fit(res_states, output_states)
+        linear_regressor.fit(res_states, output_states)
+        self.coefficients = linear_regressor.coef_
+        self.intercepts = linear_regressor.intercept_
 
-    def predict(self, input):
```

Comment:
Dims change here because sklearn regressor assumed prediction is on 2D array, but we don't use it anymore.